Dockerfile-grocy-backend

```diff
@@ -16,31 +18,40 @@ RUN set -eux; \
 # Install build-time dependencies
 RUN     apk update && \
         apk add --no-cache \
-            composer \
             git \
             gnupg \
             wget
 
 # Install system dependencies
 RUN     apk add --no-cache \
-            php7-ctype \
-            php7-fpm \
-            php7-exif \
-            php7-fileinfo \
-            php7-gd \
-            php7-iconv \
-            php7-json \
-            php7-ldap \
-            php7-pdo_sqlite \
-            php7-simplexml \
-            php7-tokenizer
+            php8 \
+            php8-ctype \
+            php8-fpm \
+            php8-exif \
+            php8-fileinfo \
+            php8-gd \
+            php8-iconv \
+            php8-ldap \
+            php8-pdo_sqlite \
+            php8-simplexml \
+            php8-tokenizer \
+            php8-phar \
+            php8-curl \
+            php8-mbstring \
+            php8-openssl \
+            php8-zip \
+            php8-intl && \
+        ln -s /usr/bin/php8 /usr/bin/php && \
```

Comment:
Perhaps it's worth separating out the PHP8 steps from the rest of the `Install system dependencies` command, and adding comment(s) to explain why we do this?

My guess / expectation is that we'll later refactor this block again, once PHP8 + Composer are available as packaged dependencies in a future version of Alpine Linux.

Reply:
Yes, sounds good to me. I've just pushed the changes.

I'm still thinking of the old docker recommendations to have as few as possible docker layers.

Reply:
Thanks - and yep, there's a trade-off to be made there related to the number of layers and caching.  That said, I think that these three layers are fairly closely related (all PHP dependency-related), and we're not doing a high volume of builds, so clarity of intent wins for now I think.

Reply:
(NB: some further small adjustments applied to the comment text for brevity)